tests/benchmarks/test_competitor_adapters.py

```
@@ -0,0 +1,1128 @@
+
+
+def test_validate_competitor_eligibility_rejects_missing_pin() -> None:
+    entry = _valid_competitor_entry()
+    del entry["pin"]
+
+    with pytest.raises(BenchmarkValidationError, match="pin"):
+        validate_competitor_eligibility(entry)
+
+
+def test_validate_competitor_eligibility_rejects_unrecognized_pin_kind() -> None:
+    with pytest.raises(BenchmarkValidationError, match="pin.kind"):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)